pkg/ipcache/ipcache.go

```diff
@@ -128,6 +129,11 @@ type IPCache struct {
 	// references to identities and removing the corresponding IPCache
 	// entries if unused.
 	deferredPrefixRelease *asyncPrefixReleaser
+
+	// labelsInjected is closed when the first label injection is complete.
+	labelsInjected       chan struct{}
+	labelsInjectedClosed bool      // because closing channels is not idempotent
```

Comment:
nit: The bool might be tempting to use but it's not using atomics or some other thread-safe accessor. Should we document this or swap this one also to a `sync.Once`?

```suggestion
	labelsInjectedClosed bool      // because closing channels is not idempotent. Used only by doLabelInjection().
```

Reply:
Ah I guess from the test code you'd need to reset the `sync.Once` to make it "more than once" if following that approach :)